includes/Classifai/Providers/OpenAI/DallE.php

```diff
@@ -52,22 +52,48 @@ public function __construct( $service ) {
 	}
 
 	/**
-	 * Register what we need for the provider.
+	 * Determine if the current user can access a feature
 	 *
-	 * This only fires if can_register returns true.
+	 * @param {string} $feature Feature to check.
+	 *
+	 * @return bool
 	 */
-	public function register() {
+	public function user_can_access( $feature ) {
+		$access   = false;
 		$settings = $this->get_settings();
 
 		// Check if the current user has permission to generate images.
 		$roles      = $settings['roles'] ?? [];
 		$user_roles = wp_get_current_user()->roles ?? [];
 
-		if (
-			current_user_can( 'upload_files' )
+		if ( current_user_can( 'upload_files' )
 			&& ( ! empty( $roles ) && empty( array_diff( $user_roles, $roles ) ) )
-			&& ( isset( $settings['enable_image_gen'] ) && 1 === (int) $settings['enable_image_gen'] )
-		) {
+			&& ( isset( $settings[ $feature ] ) && 1 === (int) $settings[ $feature ] ) ) {
```

Comment:
Similar here, I'd move this enabled check out of this method

Reply:
@dkotter Could you point me in the right direction for the `is_configured` 
 method?

cc @jeffpaul